### PR TITLE
Edit Elastic Agent docs

### PR DESCRIPTION
**Pull request by dedemorton**
Summary of changes:
- Break content into separate files so that there is a one to one mapping between the source and html file.
- Clean up asciidoc coding
- Change heading levels because we want to control nesting from the TOC rather than section levels (makes it easier to move content around and contributors don't need to worry about heading levels).
- More edits...moved some things around to improve the flow of ideas.
- Changed the name of the top-level container because I think it's weird to have two getting started guides in the same doc.

Here's what the TOC looks like when this is hooked into the nav (agent will resolve to Elastic Agent):

image

### CODE REVIEW

**Comment by dedemorton** [comment]
cc @bmorelli25 In case you want to review. Progress over perfection here....

File: x-pack/elastic-agent/docs/elastic-agent-configuration-example.asciidoc [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+experimental[]
+
+The following example shows a full list of configuration options:
+
+//REVIEWERS: I'm not sure whether this is a true statement ^^.
+
+[source,yaml]
+----
+include::elastic-agent_configuration_example.yml[]
+----
```
Comment (ph) [inline x-pack/elastic-agent/docs/elastic-agent-configuration-example.asciidoc:13]: Lets not include that file, I do not think it's up to date, I have added a task to review it.. cc @ruflin

File: x-pack/elastic-agent/docs/elastic-agent-configuration.asciidoc [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+To enable monitoring, set `settings.monitoring.enabled` to `true`. Also set the
+`logs` and `metrics` settings to control whether logs, metrics, or both are
+collected. If neither setting is specified, monitoring is disabled. Set
+`use_output` to specify the output to which monitoring events are sent.
+
+[[elastic-agent-datasource-configuration]]
+== Datasource settings
+
+//TODO: provide more context here about what a data source is and how it's used.
```
Comment (ph) [inline x-pack/elastic-agent/docs/elastic-agent-configuration.asciidoc:86]: Datasource will be removed in 7.9, so we can be high level here.

**Comment by mostlyjason** [comment]
@dedemorton can we move or link the limitations of this release to the top of each getting started guide? In particular, customers must only use it in a dedicated test cluster. If they add it to an existing cluster it could break on upgrade.

File: x-pack/elastic-agent/docs/elastic-agent-configuration-example.asciidoc (hunk 1 above)
Comment (dedemorton) [reply x-pack/elastic-agent/docs/elastic-agent-configuration-example.asciidoc:13]: I'll comment it out for now.

File: x-pack/elastic-agent/docs/elastic-agent-configuration.asciidoc (hunk 2 above)
Comment (dedemorton) [reply x-pack/elastic-agent/docs/elastic-agent-configuration.asciidoc:86]: Good. I found the concept to be a bit nebulous. Fewer concepts = easier to understand. yay!

Comment (ph) [reply x-pack/elastic-agent/docs/elastic-agent-configuration.asciidoc]: You were not alone :) cc @ruflin

**Pull request merged**